rex_xai/distributions.py

```diff
@@ -53,33 +64,21 @@ def str2distribution(d: str) -> Distribution:
 
 def random_coords(d: Optional[Distribution], *args, map=None) -> int:
     """generates random coordinates given a distribution and args"""
+
+    if args[0] < 2:
+        return -1
```

Comment:
I realise this is following how the existing code works, so feel free to disregard this comment! But I generally prefer not to return a "magic number" like this, as it can be unclear how this should be interpreted. In this case maybe returning `None` would be suitable, looking at how the return value is used in `__2d_parts`. Although a `-1` return value looks like a usable/valid value for `__1d_parts`, so perhaps it is better to return that from this function rather than adding a check for `None` in `__1d_parts`.

Reply:
I've no horror of magic numbers (very C!) but we should at least be consistent. The only reason I prefer -1 is that Python returns None from functions which do not have a formal return, whereas it'll never return -1 accidentally. It's a certain signal, rather than a programming error. We could have an enum for ReXError which would be unambiguous

Reply:
Fair enough, that reasoning makes sense to me! I like the idea of defining a ReXError - I tend to default to raising `ValueError` by default but that's not ideal.

And just to confirm, the -1 value is intended to be used in `__1d_parts`? There's no check for it there

Reply:
I can't remember about the -1 in 1d_parts. I'll have to think about it. We should definitely have a ReXError value. Happy to do this if you want, otherwise I'll leave it to you